tutor/commands/upgrade/common.py

```diff
@@ -60,6 +60,38 @@ def get_mongo_upgrade_parameters(
     return mongo_version, admin_command
 
 
+def verify_tutor_version_for_mysql_upgrade(config: Config) -> tuple[bool, str]:
```

Comment:
This function is needlessly complicated and got me really confused. What we want to do is to go through the 8.1 upgrade before the 8.4 upgrade, right? Then we only need to write:

```
def get_intermediate_mysql_upgrade(config) -> str:
    if not get_typed(config, "RUN_MYSQL", bool):
         return None
    return "docker.io/mysql:8.1.0" if get_typed(config, "DOCKER_IMAGE_MYSQL", str).startswith("docker.io/mysql:8.4") else None
```

Reply:
The condition was checking against two specific MySQL versions was in the case someone jumps multiple tutor versions, e.g, tutor 17 -> 20 and the new MySQL image is on 9.0 for example, this upgrade step wouldn't run and the upgrade process would crash again. But I think that would become too complex and this solution is more elegant. 

Should we not echo the info that MySQL is not running and it is up to the user to upgrade it?

Reply:
> The condition was checking against two specific MySQL versions was in the case someone jumps multiple tutor versions, e.g, tutor 17 -> 20 and the new MySQL image is on 9.0 for example

In that case, the condition should be: 

    import packaging
    target_version = packaging.version(get_typed(config, "DOCKER_IMAGE_MYSQL", str))
    return "docker.io/mysql:8.1.0" if target_version > packaging.version("8.1.0") else None

Reply:
> target_version = packaging.version(get_typed(config, "DOCKER_IMAGE_MYSQL", str))

This will fail if the image tag is set to latest. [docs reference](https://packaging.pypa.io/en/stable/version.html#packaging.version.InvalidVersion)

Reply:
Obviously. So what should we do in that case?

Reply:
I've added a default check for latest as it would always be greater than 8.1 for our case.